### PR TITLE
Naming, immutable array manipulation

### PR DESCRIPTION
**Pull request by cboden**

### CODE REVIEW

**Comment by clue** [comment]
:+1:

**Comment by WyriHaximus** [comment]
:+1: thanks!

File: src/RequestParser.php [hunk 1]
```diff
@@ -99,12 +99,13 @@ public function parseHeaders($data)
             parse_str($queryString, $parsedQuery);
         }
 
-        $headers = $psrRequest->getHeaders();
-        array_walk($headers, function(&$val) {
+        $headers = array_map(function(&$val) {
```
Comment (elazar) [inline src/RequestParser.php]: `$val` doesn't still need to be passed by reference here, does it?

**Comment by cboden** [comment]
Ooops. I copy/paste fail.

**Comment by WyriHaximus** [comment]
Do you want to PR and fix it or want me to?

**Comment by cboden** [comment]
I'd say just go ahead and push the fix :)

**Comment by WyriHaximus** [comment]
Done :+1:

**Pull request merged**